crates/biome_js_formatter/src/js/expressions/call_arguments.rs

```diff
@@ -667,12 +668,43 @@ fn with_token_tracking_disabled<F: FnOnce(&mut JsFormatter) -> R, R>(
     result
 }
 
-/// Tests if `expression` has an empty parameters list.
-fn has_no_parameters(expression: &JsFunctionExpression) -> bool {
-    match expression.parameters() {
-        // Use default formatting for expressions without parameters, will return `Err` anyway
-        Err(_) => true,
-        Ok(parameters) => parameters.items().is_empty(),
+/// Tests if all of the parameters of `expression` are simple enough to allow
+/// a function to group.
+fn has_only_simple_parameters(expression: &JsFunctionExpression) -> bool {
+    expression.parameters().map_or(true, |parameters| {
+        for parameter in parameters.items().into_iter().flatten() {
+            if !is_simple_parameter(parameter) {
+                return false;
+            }
+        }
+
+        true
```

Comment:
Do you think this coudl be simplified to the following?

We often use a fp-style in the code base.

```suggestion
        parameters.items().into_iter().flatten().all(is_simple_parameter)
```